tests/test_daily_temperature_extrema.py

```diff
@@ -0,0 +1,543 @@
+import datetime
+
+from pytest import fixture, mark, approx
+from sqlalchemy.sql import text
+
+from pycds.util import generic_sesh
+from pycds import Network, Station, History, Variable, Obs, NativeFlag, PCICFlag
+from pycds.weather_anomaly import DailyMaxTemperature, DailyMinTemperature
+
+
+def describe_function_effective__day__for__Tmax():
+    @mark.parametrize('tod, obs_time', [
+        ('morning', '2000-01-01 07:23'),
+        ('afternoon', '2000-01-01 16:18')
+    ])
+    @mark.parametrize('freq, expected_day', [
+        ('1-hourly', {'morning': datetime.datetime(2000, 1, 1), 'afternoon': datetime.datetime(2000, 1, 1)}),
+        ('12-hourly', {'morning': datetime.datetime(2000, 1, 1), 'afternoon': datetime.datetime(2000, 1, 2)})
+    ])
+    def it_returns_the_expected_day_of_observation(mod_empty_database_session, tod, obs_time, freq, expected_day):
+        result = mod_empty_database_session.execute(
+            text('SELECT effective_day_for_Tmax(:obs_time, :freq) AS eday'),
+            {'obs_time': obs_time, 'freq': freq}
+        ).fetchall()
+        assert len(result) == 1
+        assert result[0]['eday'] == expected_day[tod]
+
+
+views = [DailyMaxTemperature, DailyMinTemperature]
+
+@fixture(scope='module')
+def with_views_sesh(mod_empty_database_session):
+    sesh = mod_empty_database_session
+    for view in views:
+        view.create(sesh)
+    yield sesh
+    for view in reversed(views):
+        view.drop(sesh)
+
+
+def refresh_views(sesh):
+    for view in views:
+        view.refresh(sesh)
+
+
+def describe_with_1_network():
```

Comment:
I added a long section in the README explaining the test framework. I think part of the issue is just familiarity with this style of testing. To me it reads fairly straightforwardly (says the author, always :)), though it is fair comment that it can take some effort to trace the specific context of any given test. That is partly why I wrote the output formatter - it provides a succinct summary of the tests in a highly readable format. Supposing that your `describe_` functions are accurately named (i.e., correspond to the test condition set up in the fixture), the formatted test output is a great way to understand what is going on.